### ISSUE TITLE
debugger doesn't seem to work with functions w/source refs?

### ISSUE THREAD

**Issue by kevinushey** [rstudio/rstudio#7749]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.4.787
    OS Version      : macOS 10.15.6
    R Version       : 4.0.2

### Steps to reproduce the problem

Create an R file with the contents:

```
foo <- function(x) {
   print(1)
   print(2)
   print(3)
}

foo(1)  
```

Save the file, set a breakpoint on line 3, and try to source the document.

### Describe the problem in detail

The debugger is started, but the source editor does not update to match the current context. In addition, typing `n` to step forward causes focus to be sent to the Source pane, making interactive debugging a pain.

### Describe the behavior you expected

The IDE should update to show the debugged context, and focus should not be sent to the Source pane.

- [x] I have read the guide to submitting good bug reports at https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports . 
- [x] I have installed the latest version of RStudio and confirmed that the issue still persists.
- [x] If I am reporting a RStudio crash, I have included a diagnostics report. https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kevinushey** [comment]
I'm realizing now that the cursor position is being set to the right debug location, but the debug highlight that's normally applied is not being used here. Note that there are two changes in behavior here:

1. Previously, we did not update the cursor position when debugging; we only highlighted the current line being executed.
2. Focus remained on the Console pane while debugging.

**Comment by jmcphers** [comment]
@kevinushey Does #7751 fix this?

**Comment by kevinushey** [comment]
Indeed it appears to have been fixed by #7751!

**Comment by dfalty** [comment]
Verified fixed using the same steps listed. version 1.4.874
